core/src/test/java/org/apache/iceberg/TestPartitionSpecBuilderCaseSensitivity.java

```diff
@@ -0,0 +1,697 @@
+          required(1, "id", Types.IntegerType.get()),
+          required(2, "data", Types.StringType.get()),
+          required(3, "category", Types.StringType.get()),
+          required(4, "order_date", Types.DateType.get()),
+          required(5, "order_time", Types.TimestampType.withoutZone()),
+          required(6, "ship_date", Types.DateType.get()),
+          required(7, "ship_time", Types.TimestampType.withoutZone()));
+
+  private static final Schema SCHEMA_CASE_SENSITIVE =
+      new Schema(
+          required(1, "id", Types.IntegerType.get()),
+          required(2, "data", Types.StringType.get()),
+          required(3, "DATA", Types.StringType.get()),
+          required(4, "order_date", Types.DateType.get()),
+          required(5, "ORDER_DATE", Types.DateType.get()),
+          required(6, "order_time", Types.TimestampType.withoutZone()),
+          required(7, "ORDER_TIME", Types.TimestampType.withoutZone()));
+
+  @TempDir private Path temp;
+  private File tableDir = null;
+
+  @BeforeEach
+  public void setupTableDir() throws IOException {
+    this.tableDir = Files.createTempDirectory(temp, "junit").toFile();
+  }
+
+  @AfterEach
+  public void cleanupTables() {
+    TestTables.clearTables();
+  }
+
+  @Test
+  public void testPartitionTypeWithColumnNamesThatDifferOnlyInLetterCase() {
+    Schema schema =
+        new Schema(
+            required(1, "id", Types.IntegerType.get()),
+            required(2, "data", Types.StringType.get()),
+            required(3, "DATA", Types.StringType.get()),
+            required(4, "order_date", Types.DateType.get()));
+    PartitionSpec spec = PartitionSpec.builderFor(schema).identity("data").identity("DATA").build();
+    TestTables.TestTable table =
+        TestTables.create(tableDir, "test", schema, spec, V2_FORMAT_VERSION);
+
+    StructType expectedType =
+        StructType.of(
+            NestedField.optional(1000, "data", Types.StringType.get()),
+            NestedField.optional(1001, "DATA", Types.StringType.get()));
+    StructType actualType = Partitioning.partitionType(table);
+    assertThat(actualType).isEqualTo(expectedType);
+  }
+
+  @Test
+  public void testPartitionTypeWithIdentityTargetName() {
+    PartitionSpec spec =
+        PartitionSpec.builderFor(SCHEMA_CASE_INSENSITIVE).identity("data", "p1").build();
+    TestTables.TestTable table =
+        TestTables.create(tableDir, "test", SCHEMA_CASE_INSENSITIVE, spec, V2_FORMAT_VERSION);
+
+    StructType expectedType =
+        StructType.of(NestedField.optional(1000, "p1", Types.StringType.get()));
+    StructType actualType = Partitioning.partitionType(table);
+    assertThat(actualType).isEqualTo(expectedType);
+  }
+
+  @Test
+  public void testBucketSourceNameDoesNotAllowExactDuplicateWhenCaseSensitive() {
+    assertThatIllegalArgumentException()
+        .isThrownBy(
+            () ->
+                PartitionSpec.builderFor(SCHEMA_CASE_SENSITIVE)
+                    .bucket("data", 10, "p1")
```

Comment:
nit: it is a bit difficult to read `p1` vs `P1`. use some other more distinguiable names, like `partition1` and `PARTITION1`. this applies to other tests too

Reply:
fixed